crates/wasi/src/preview2/poll.rs

```diff
@@ -121,28 +122,81 @@ impl<T: WasiView> poll::Host for T {
             }
         }
 
-        Ok(PollOneoff {
+        Ok(PollList {
             elems: closure_futures,
         }
         .await?)
     }
+
+    async fn poll_one(&mut self, pollable: Resource<Pollable>) -> Result<()> {
+        let table = self.table_mut();
+
+        let closure_future: PollableFuture<'_>;
+        match table.get_host_pollable_mut(&pollable)? {
+            HostPollable::Closure(f) => closure_future = f(),
+            HostPollable::TableEntry { index, make_future } => {
+                closure_future = make_future(index);
+            }
+        }
+
+        struct PollOne<'a> {
+            elem: PollableFuture<'a>,
+        }
+        impl<'a> Future for PollOne<'a> {
+            type Output = Result<()>;
+
+            fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+                match self.elem.as_mut().poll(cx) {
+                    Poll::Ready(Ok(())) => Poll::Ready(Ok(())),
+                    Poll::Ready(Err(e)) => Poll::Ready(Err(e.context("poll_one"))),
+                    Poll::Pending => Poll::Pending,
+                }
+            }
+        }
+
+        Ok(PollOne {
+            elem: closure_future,
+        }
+        .await?)
+    }
```

Comment:
I think this can all be skipped in favor fo `closure_future.await`?

`poll-list` is a bit funky since there's more than one future, but here there's just one so I think it should be ok to do a normal `.await`

Reply:
Done.